src/Simplex/Messaging/Agent.hs

```diff
@@ -1445,16 +1453,26 @@ getConnectionRatchetAdHash' c connId = do
 
 connectionStats :: Connection c -> ConnectionStats
 connectionStats = \case
-  RcvConnection ConnData {ratchetSyncState} rq ->
-    ConnectionStats {rcvQueuesInfo = [rcvQueueInfo rq], sndQueuesInfo = [], ratchetSyncState}
-  SndConnection ConnData {ratchetSyncState} sq ->
-    ConnectionStats {rcvQueuesInfo = [], sndQueuesInfo = [sndQueueInfo sq], ratchetSyncState}
-  DuplexConnection ConnData {ratchetSyncState} rqs sqs ->
-    ConnectionStats {rcvQueuesInfo = map rcvQueueInfo $ L.toList rqs, sndQueuesInfo = map sndQueueInfo $ L.toList sqs, ratchetSyncState}
-  ContactConnection ConnData {ratchetSyncState} rq ->
-    ConnectionStats {rcvQueuesInfo = [rcvQueueInfo rq], sndQueuesInfo = [], ratchetSyncState}
-  NewConnection ConnData {ratchetSyncState} ->
-    ConnectionStats {rcvQueuesInfo = [], sndQueuesInfo = [], ratchetSyncState}
+  RcvConnection cData rq ->
+    (stats cData) {rcvQueuesInfo = [rcvQueueInfo rq]}
+  SndConnection cData sq ->
+    (stats cData) {sndQueuesInfo = [sndQueueInfo sq]}
+  DuplexConnection cData rqs sqs ->
+    (stats cData) {rcvQueuesInfo = map rcvQueueInfo $ L.toList rqs, sndQueuesInfo = map sndQueueInfo $ L.toList sqs}
+  ContactConnection cData rq ->
+    (stats cData) {rcvQueuesInfo = [rcvQueueInfo rq]}
+  NewConnection cData ->
+    stats cData
+  where
+    stats cData@ConnData {connAgentVersion, ratchetSyncState} =
+      ConnectionStats
+        { connAgentVersion,
+          rcvQueuesInfo = [],
+          sndQueuesInfo = [],
+          ratchetSyncState,
+          ratchetSyncAllowed = ratchetSyncAllowed' cData,
```

Comment:
I thought the idea is to return True here even if it's not supported, so that in this case the button would show and suggest to ask another party to update. And in this case where required you need to check for both allowed and supported.